mk/dev.mk

```diff
@@ -38,32 +38,46 @@ endef
 # so this is location should not be changed by developers.
 KUBECONFIG_DIR := $(HOME)/.kube
 
-PROTOS_DEPS_PATH=$(CI_TOOLS_DIR)/protos
-
-CLANG_FORMAT=$(CI_TOOLS_BIN_DIR)/clang-format
-YQ=$(CI_TOOLS_BIN_DIR)/yq
-HELM=$(CI_TOOLS_BIN_DIR)/helm
-K3D_BIN=$(CI_TOOLS_BIN_DIR)/k3d
-KIND=$(CI_TOOLS_BIN_DIR)/kind
-KUBEBUILDER=$(CI_TOOLS_BIN_DIR)/kubebuilder
-KUBEBUILDER_ASSETS=$(CI_TOOLS_BIN_DIR)
-CONTROLLER_GEN=$(CI_TOOLS_BIN_DIR)/controller-gen
-KUBECTL=$(CI_TOOLS_BIN_DIR)/kubectl
-PROTOC_BIN=$(CI_TOOLS_BIN_DIR)/protoc
-SHELLCHECK=$(CI_TOOLS_BIN_DIR)/shellcheck
-CONTAINER_STRUCTURE_TEST=$(CI_TOOLS_BIN_DIR)/container-structure-test
+ifeq (,$(shell which mise))
+$(error "mise - https://github.com/jdx/mise - not found. Please install it.")
+endif
+MISE := $(shell which mise)
+
+PROTOS_DEPS_PATH=$(shell $(MISE) where protoc)/include
+
+XDS_VERSION=$(shell go list -f '{{ .Version }}' -m github.com/cncf/xds/go)
+PROTO_XDS=$(shell go mod download github.com/cncf/xds@$(XDS_VERSION) && go list -f '{{ .Dir }}' -m github.com/cncf/xds@$(XDS_VERSION))
```

Comment:
I'm not a fan of these because if we don't have internet this fails.

I guess these can't work with a mise tool? Why?
Would it make sense to write a mise plugin to do this?

Reply:
Right now, mise does not have support for this. For go mise can only do `go install`, and we need to do `go mod download`

Looking at the mise plugins. I think we would need to create a custom plugin for every proto dependency. If I understand correctly, one mise plugin should install one dependency. 

I am not sure if there is an easy way of depending on protos. I see that people most of the time are just downloading this schema manually: https://github.com/search?q=github.com%2Fenvoyproxy%2Fdata-plane-api&type=code&p=1

Reply:
https://github.com/kumahq/kuma/issues/13880